src/main/java/org/padler/gradle/minify/minifier/Minifier.java

```diff
@@ -1,29 +1,44 @@
 package org.padler.gradle.minify.minifier;
 
+import org.gradle.api.GradleException;
+
 import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.nio.file.*;
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public abstract class Minifier {
 
+    protected MinifierOptions minifierOptions = new MinifierOptions();
+
+    protected final Report report = new Report();
+
     public void minify(String srcDir, String dstDir) {
+        minifyInternal(srcDir, dstDir);
+        System.err.println(createReport());
+        if (!report.getErrors().isEmpty())
+            throw new GradleException(report.getErrors() + " Errors in " + getMinifierName());
+    }
+
+    protected void minifyInternal(String srcDir, String dstDir) {
         try (Stream<Path> filesStream = Files.list(Paths.get(srcDir)).filter(f -> !f.toString().equals(srcDir))) {
             List<Path> files = filesStream.collect(Collectors.toList());
             for (Path f : files) {
                 if (f.toFile().isFile()) {
                     Path dst = Paths.get(dstDir);
-                    File dstFile = new File(dst.toString(), f.getFileName().toString());
+                    String fileName = f.getFileName().toString();
+                    if (Boolean.TRUE.equals(minifierOptions.getOriginalFileNames())) {
```

Comment:
Should be inverted because only if originalFileNames is true renaming should be skiped

Reply:
🤦‍♂️
fixed